src/backend/cpu/kernel/wrap.hpp

```diff
@@ -25,6 +29,8 @@ void wrap_dim(Param<T> out, CParam<T> in, const dim_t wx, const dim_t wy,
     af::dim4 istrides = in.strides();
     af::dim4 ostrides = out.strides();
 
+    std::fill(begin(out), end(out), scalar<T>(0));
```

Comment:
I don't like this. the out array should be zeroed before we get here. Instead of createHandle, you should use createValueArray.

If the out array is passed into the function, we can stipulate that the values will be summed into the output array. If they want the originals they need to pass a constant array or zero beforehand.